django_tutorial/first_project/README.md

```diff
@@ -8,17 +8,53 @@ The code is split into 2 apps:
 - `web_app` contains some static HTML/CSS/JS running a rich client webapp that consumes the API and renders the visualization
 
-## Running it
+## Web app directory and build structure
 
-For anyone wondering how to run this locally, you can do the following:
+Here's a little more info about the frontend build and toolchain, in case anyone is new to it.
+
+### Build tools
+
+We use NPM to run Webpack, which transforms and bundles the source JS files from `static_src` and puts them in the `static` directory.  Webpack uses a number of plugins to transform the JS; one of these plugins is Babel, which is configured to convert JSX syntax into JS, and convert ES2015 features down to ES5 which all browsers can handle.  You can see the NPM configuration in `package.json`, the Webpack configuration in `webpack.config.js`, and the Babel configuration in `.babelrc` (all in the `web_app` directory).
+
+### Directories
+
+- `templates` contains the HTML templates; note they're grouped into a `web_app` subdirectory to avoid confusion in Django's template lookup
+- `static_src` contains any source JS/CSS image files that will be processed and bundled by Webpack/Babel.  If you want to make changes to the frontend, this is the place to do it.
+- `static` contains the static JS/CSS/image files that will be served by the webserver.  Webpack puts its output bundles here.  Note you should NOT make any manual changes to the files in this directory, as they are auto-generated by the build process and are in `.gitignore` anyway.
+
+### NPM Scripts
+
+You can use NPM to run the following scripts:
+1. `npm run build`: builds the JS bundles for local development
+1. `npm run build-production`: builds the JS bundles for production deployment (eg may include minimization)
+1. `npm run build:watch`: builds the JS bundles for local development, and then continues to watch the JS source files so any changes to them are automatically rebuilt into the JS bundles
+1. `npm run test`: runs the JS unit tests
+1. `npm run test:watch`: runs the JS unit tests and continues to watch the JS files so the tests are re-run whenever the files change
+
+Also, look at `web_app/package.json` to see exactly what these scripts are doing
+
+## Running the webapp locally
+
+For anyone wanting to run the whole app locally, you can do the following.
+
+
+Run the following commands once, to get everything set up:
 
 1. Set up a local DB and populate it with data (either by dumping the prod DB or by running the pipeline locally, as described [here](../../pipeline/README.md))
 1. Set the environment variable for your database URL, e.g. `export SBA_DWH='postgresql://username:password@127.0.0.1:5432/postgres'`
-1. Run `python manage.py runserver` from within this directory
+1. Install [Node and NPM](https://www.npmjs.com/get-npm)
+1. From within the `web_app` directory, run `npm install` to load all the required NPM packages listed in `package.json`
```

Comment:
I wonder if it's possible to use anaconda to manage `npm` stuff: https://anaconda.org/cpcloud/npm